### PR TITLE
Infotip for small screens

### PR DESCRIPTION
**Pull request by raveenakothapally**
# InfoTips
Infotips activated when engaged by the user. Tapping or clicking the info bubble initiates flyout. On small screens, it's a modal.

Kapture 2020-05-06 at 9 18 39

### CODE REVIEW

File: src/less/tooltip/base/tooltip.less [hunk 1]
```diff
@@ -162,6 +162,66 @@ span.tourtip {
...
+.dialog-infotip__header {
+    .dialog-header-content(0);
+
+    align-items: flex-start;
+}
+
+.dialog-infotip__close {
+    background-color: @dialog-back-close-background-color;
+    border: 0;
+    border-top-right-radius: 7px;
```
Comment (agliga) [inline src/less/tooltip/base/tooltip.less:199]: If we change the `margin` to be greater than `7px`, then this isn't needed. But then the `padding` will need to be reduced to compensate
So maybe `padding: 8px; margin: 8px;` works

File: src/less/tooltip/base/tooltip.less [hunk 2]
```diff
@@ -162,6 +162,66 @@ span.tourtip {
...
+    padding-bottom: 15vh;
+    top: 15%;
+    width: calc(100% - 32px);
+
+    &::before {
+        border-radius: 7px;
+    }
+}
+
+.dialog-infotip__window:not(.dialog-infotip__window--full) {
```
Comment (agliga) [inline src/less/tooltip/base/tooltip.less:181]: This is a weird way to do this. Can you not do the `dialog-infotip__window--full` as a separate class and have those styles there? 
So the base should have all the styles it needs, and the full should have what it needs (and override if needed). 
Remember in CSS order matters to in terms of overriding styles. 

Also there is no full example

File: src/less/tooltip/base/tooltip.less [hunk 3]
```diff
@@ -162,6 +162,66 @@ span.tourtip {
...
 
+.dialog-infotip[role="dialog"] {
+    .dialog-base();
+}
+
+.dialog-infotip__window {
+    .dialog-window();
+
+    padding-bottom: 15vh;
+    top: 15%;
```
Comment (agliga) [inline src/less/tooltip/base/tooltip.less:173]: This is getting overriden by the `top:50` in the the other styles, is it needed?

File: src/less/tooltip/base/tooltip.less [hunk 4]
```diff
@@ -162,6 +162,66 @@ span.tourtip {
...
+    &:focus {
+        color: @dialog-back-close-hover-focus-color;
+    }
+}
+
+.dialog-infotip__main {
+    .dialog-body-content();
+
+    margin-right: 0;
+    min-height: 18px;
```
Comment (agliga) [inline src/less/tooltip/base/tooltip.less:216]: Why is the `min-height` needed? I removed it and was working fine.

File: src/less/tooltip/base/tooltip.less (hunk 1 above)
Comment (raveenakothapally) [reply src/less/tooltip/base/tooltip.less:199]: Thanks for the tip!

File: src/less/tooltip/base/tooltip.less [hunk 5]
```diff
@@ -162,6 +162,66 @@ span.tourtip {
...
+    }
+}
+
+.dialog-infotip__window:not(.dialog-infotip__window--full) {
+    flex-direction: row-reverse;
+    left: 50%;
+    max-width: 400px;
+    min-width: 200px;
+    top: 50%;
+    transform: translate(-50%, -25%);
```
[the remark this answers is not in the document]
Comment (raveenakothapally) [reply src/less/tooltip/base/tooltip.less:187]: To center the dialog horizontal and vertical

**Comment by raveenakothapally** [comment]
> I looked at it purely form a code/functional point of view. I did not look at the spec and see how well it matches.
> Overall nice job, I think the main comments were mainly nits, but the overall code looks great and is exactly what I was envisioning.
> I almost feel like it belongs as a separate module although I'll let Ian make that final call.

Thanks for the review @agliga. Addressed your concerns; could you please take another look?
The left and transform is done in order to center the dialog horizontally and vertically (Spec wants this to be centered). Please let me know if you have any other suggestions to achieve this behavior.

File: src/less/tooltip/base/tooltip.less (hunk 5 above)
[the remark this answers is not in the document]
Comment (agliga) [reply src/less/tooltip/base/tooltip.less:177]: So that's fine, I'm just wondering why it's `-25%` and not `-50%` to center it 
https://css-tricks.com/centering-css-complete-guide/ (In the centering both vertically and horizontally and unknown width/height)

File: docs/_includes/common/tooltip.html [hunk 6]
```diff
@@ -181,6 +181,33 @@ <h3 class="infotip__heading">Infotip</h3>
 </p>
     {% endhighlight %}
 
+    <p>Infotip on small screens</p>
```
Comment (ianmcburnie) [inline docs/_includes/common/tooltip.html:184]: @raveenakothapally Let's also add this demo and code example under the dialog section. Before release I will most likely rename it something a little more generic than `dialog-infotip`, because I see it's potential in places other than infotip, but the current name will suffice for now.

File: src/less/tooltip/base/tooltip.less [hunk 7]
```diff
@@ -162,6 +162,58 @@ span.tourtip {
     width: @icon-width;
 }
 
+.dialog-infotip[role="dialog"] {
```
Comment (ianmcburnie) [inline src/less/tooltip/base/tooltip.less:165]: We should move all of this `dialog-infotip` under the dialog module. Infotip will then have a dependency on the dialog module.

File: src/less/tooltip/base/tooltip.less (hunk 5 above)
[the remark this answers is not in the document]
Comment (raveenakothapally) [reply src/less/tooltip/base/tooltip.less:177]: I used that same guideline but `-50%` didn't exactly center it :)

File: dist/drawer/ds6/drawer.css [hunk 8]
```diff
@@ -91,10 +91,6 @@
 .drawer__window {
   display: -webkit-box;
   display: flex;
-  -webkit-box-orient: vertical;
```
Comment (ianmcburnie) [inline dist/drawer/ds6/drawer.css:94]: Is this change intentional? Looks like it has broken the drawer:

Screen Shot 2020-05-05 at 9 05 33 PM

File: src/less/dialog/base/dialog.less [hunk 9]
```diff
@@ -186,12 +188,63 @@
...
+    .dialog-header-content(0);
+
+    align-items: flex-start;
+}
+
+.dialog-infotip__close {
+    background-color: @dialog-back-close-background-color;
+    border: 0;
+    color: @dialog-back-close-color;
+    cursor: pointer;
```
Comment (ianmcburnie) [inline src/less/dialog/base/dialog.less:221]: Nitpick, but please remove `cursor: pointer`. For a11y reasons we only have pointer on links, not buttons. Also I don't think `z-index` is needed...?

Comment (raveenakothapally) [reply src/less/dialog/base/dialog.less:221]: ACK on `cursor`

`z-index` is something that is copied over from `dialog__close`. Not sure why it was needed back during dialog impl. Could be to give precedence (Top of) over default elements just in case.

File: dist/drawer/ds6/drawer.css (hunk 8 above)
Comment (raveenakothapally) [reply dist/drawer/ds6/drawer.css:94]: I searched for `.dialog-window()` and it did not give a usage in drawer (As it uses `.dialog-window(100%)`) so missed updating `drawer` module. Thanks for catching this bug. This should be fixed now.

**Comment by ianmcburnie** [comment]
Going with #1100 instead. Closing.
